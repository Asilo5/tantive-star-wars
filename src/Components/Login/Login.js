import React, { Component } from 'react';
import './Login.scss';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class Login extends Component {
    constructor() {
        super();
        this.state = {
            name: '',
            quote: '',
            rank: ''
        }
    }

    handleChange = (e) => {
       this.setState({ [e.target.name] : e.target.value })
    }

    sendUpState = () => {
       const { userInfo } = this.props;
       userInfo(this.state);
    }

    render() {
        const { name, rank, quote } = this.state;
        return (
          <div>
            <h1>Star Wars Trivia</h1>
            <img className="yoda_head" src="http://pluspng.com/img-png/yoda-head-png-download-800.gif" alt="yoda's face"></img>
            <h2 className="tantive_heading">Tantive Edition</h2>
            <form>
                <input className='user-name'
                       placeholder='Must Name Here'
                       type='text' 
                       name='name'
                       value={this.state.name} 
                       onChange={(e) => this.handleChange(e)}/>
                <input className='user-quote'
                       placeholder='Quote Favourite'
                       type='text' 
                       name='quote'
                       value={this.state.quote} 
                       onChange={(e) => this.handleChange(e)}/>
                <p>What Knowledge have you from Star Wars?</p>
                <section>
                    <button  
                      className="knowledge_button" 
                      type='button' 
                      name='rank' 
                      value='Novice' 
                      onClick={(e) => this.handleChange(e)} 
                      >Novice
                    </button>
                    <button 
                      className="knowledge_button" 
                      type='button' 
                      name='rank' 
                      value='Intermediate' 
                      onClick={(e) => this.handleChange(e)}>Intermediate
                    </button>
                    <button 
                      className="knowledge_button"  
                      type='button' 
                      name='rank' 
                      value='Expert' 
                      onClick={(e) => this.handleChange(e)}
                      >Expert
                    </button>
                </section>
                { name === '' || quote === '' || rank === '' ? 
                ( <button className='submit-btn-error' disabled={true}> Inputs Must Fill </button>)
                : 
                 (<Link to='/movies'><button className='submit-btn' onClick={this.sendUpState}>Submit</button></Link> )
                }
            </form>
          </div>
        )
    }
}

Login.propTypes = {
  userInfo: PropTypes.func
}

export default Login;