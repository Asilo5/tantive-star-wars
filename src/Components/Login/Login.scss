form {
    background: rgba(255, 255, 255, 0.322);
    border: 1px solid red;
    width: 50%;
    margin: auto;
    border-radius: 20px;
    display: flex;
    flex-direction: column;
    margin-top: 2em;
    padding: 30px;
    input {
        font-size: 2.2em;
        margin: 10px auto;
        width: 92%;
        text-align: center;
        border: none;
        caret-color: red;
        border-bottom: 2px solid red;
        ::placeholder {
           text-align: center;
        }
    }
    p {
      font-size: 2em;
      font-weight: bold;
      color: red;
    }
    button {
        font-size: 3em;
        margin: 5px;
        padding: 13px;
        border-radius: 10px;
        border: 2px solid transparent;
        cursor: pointer;
        &:hover,
        &:focus {
           border: 2px solid red;
           color: red;
        }
    }
    .submit-btn {
        font-size: 3em;
        width: 30%;
        margin: 10px auto;
        padding: 10px;
        text-decoration: none;
        color: red;
        border-radius: 10px;
        &:hover {
            border: 2px solid red;
        }
    }
    .submit-btn-error {
      @extend .submit-btn;
      width: 45%;
      cursor: not-allowed;
      color: gray;
      &:hover {
        border: 2px solid transparent;
        color: gray;
    }
    }
}

section {
    display: flex;
}

body {
    background-image: url('https://media.giphy.com/media/3ov9k1173PdfJWRsoE/giphy.gif');
    background-size: cover;
    background-repeat: no-repeat;
    height: 100vh;
    width: 100%;
    text-align: center;
}

h1 {
    color: white;
    font-size: 8em;
}

