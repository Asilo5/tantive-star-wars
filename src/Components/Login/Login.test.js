import React from 'react';
import { shallow } from 'enzyme';
import Login  from './Login';

describe('Login', () => {

    let wrapper;
    let mockUserInfo = jest.fn();

    beforeEach(() => {
        wrapper=shallow(<Login userInfo={mockUserInfo}/>)
    })

    it('should match snapshot with all information passing in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should call handleChange when input changes', () => {
      
    })

    it('should update state when handleChange is called', () => {

    })

    it('should invoke handleChange on click', () => {
        
    })

    it('should call userInfo when sendUpState is invoked', () => {

    })

    it('should invoke sendUpdate on click', () => {

    })
})