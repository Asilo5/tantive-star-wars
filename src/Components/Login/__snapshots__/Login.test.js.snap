// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot with all information passing in correctly 1`] = `
<form>
  <input
    className="user-name"
    name="name"
    onChange={[Function]}
    placeholder="Must Name Here"
    type="text"
    value=""
  />
  <input
    className="user-quote"
    name="quote"
    onChange={[Function]}
    placeholder="Quote Favourite"
    type="text"
    value=""
  />
  <p>
    What Knowledge have you from Star Wars?
  </p>
  <section>
    <button
      name="rank"
      onClick={[Function]}
      type="button"
      value="Novice"
    >
      Novice
    </button>
    <button
      name="rank"
      onClick={[Function]}
      type="button"
      value="Intermediate"
    >
      Intermediate
    </button>
    <button
      name="rank"
      onClick={[Function]}
      type="button"
      value="Expert"
    >
      Expert
    </button>
  </section>
  <button
    className="submit-btn-error"
    disabled="true"
  >
     Inputs Must Fill 
  </button>
</form>
`;
